lib/facter/facts/azurelinux/os/release.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Facts
+  module Azurelinux
+    module Os
+      class Release
+        FACT_NAME = 'os.release'
+        ALIASES = %w[operatingsystemmajrelease operatingsystemrelease].freeze
+
+        def call_the_resolver
+          version = from_specific_file || from_os_release
+
+          return Facter::ResolvedFact.new(FACT_NAME, nil) unless version
+
+          [Facter::ResolvedFact.new(FACT_NAME, version),
+           Facter::ResolvedFact.new(ALIASES.first, version['major'], :legacy),
+           Facter::ResolvedFact.new(ALIASES.last, version['full'], :legacy)]
+        end
+
+        def from_specific_file
+          version = Facter::Resolvers::SpecificReleaseFile.resolve(:release,
+                                                                   { release_file: '/etc/azurelinux-release',
```

Comment:
Should we be using the /etc/os-release which is the standard for systemd based system?

Reply:
We could, but the base linux resolver does not parse the full version correctly so I parse `/etc/azurelinux-release` similarly to what I did with mariner parsing `/etc/mariner-release` in PR https://github.com/puppetlabs/facter/pull/2519

Reply:
> the base linux resolver does not parse the full version correctly

Could you expand on that? What does it parse currently and what would you expect it to?

Reply:
Sorry for the late reply, was on vacation.
With just the base resolver, we get:
```
  release => {
    full => "3.0",
    major => "3.0"
  },
```
vs. the more desirable:
```
  release => {
    full => "3.0.20240401",
    major => "3",
    minor => "0"
  },
```